src/Symfony/Component/Uid/Uuid.php

```diff
@@ -117,6 +117,14 @@ final public static function v6(): UuidV6
 
     public static function isValid(string $uuid): bool
     {
+        if (self::NIL === $uuid && \in_array(static::class, [__CLASS__, NilUuid::class], true)) {
```

Comment:
What is this `static::class` sorcery doing?

Reply:
`Uuid::isValid()` is used by all child classes. We want to allow `Uuid::isValid('0000...')` and `NilUuid::isValid('0000...')` but not `UuidV4::isValid('0000...')` for example.

Reply:
Oh, I see. Thanks for the explanation. Wouldn't it make sense to override the method in those child classes instead? Either way, the change looks good to me.